rust/signed_doc/src/builder.rs

```diff
@@ -1,15 +1,18 @@
 //! Catalyst Signed Document Builder.
+use catalyst_types::kid_uri::KidUri;
+use ed25519_dalek::{ed25519::signature::Signer, SecretKey};
+
 use crate::{CatalystSignedDocument, Content, InnerCatalystSignedDocument, Metadata, Signatures};
 
 /// Catalyst Signed Document Builder.
-#[derive(Debug, Default)]
+#[derive(Debug, Default, Clone)]
 pub struct Builder {
     /// Document Metadata
-    metadata: Option<Metadata>,
+    pub(crate) metadata: Option<Metadata>,
```

Comment:
Why we need these fields to be `pub(crate)`. I think they should private even for this crate